app/services/EventReportService.scala

```diff
@@ -187,21 +189,58 @@ class EventReportService @Inject()(eventReportConnector: EventReportConnector,
     }
   }
 
-  def submitEventDeclarationReport(pstr: String, userAnswersJson: JsValue)
-                                  (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, request: RequestHeader): Future[JsValue] = {
-    eventReportConnector.submitEventDeclarationReport(pstr, userAnswersJson).map(_.json)
+  private def validatePayloadAgainstSchema(payload: JsObject, schemePath: String, eventName: String)
+                                          (implicit ec: ExecutionContext): Future[Unit] = {
+    Future.fromTry(jsonPayloadSchemaValidator.validatePayload(payload, schemePath, eventName)).map(_ => (): Unit)
   }
 
-  def submitEvent20ADeclarationReport(pstr: String, data: JsValue)
-                                     (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, request: RequestHeader): Future[JsValue] = {
-    eventReportConnector.submitEvent20ADeclarationReport(pstr, data).map(_.json)
+  def submitEventDeclarationReport(pstr: String, userAnswersJson: JsValue)
+                                  (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, request: RequestHeader): Future[Unit] = {
+
+    def recoverAndValidatePayload(transform1828toETMP: JsObject): Future[Unit] = {
```

Comment:
Is `transform1828toETMP` a good name for this parameter? Isn't this just the payload?

Reply:
Changed